fhir-model/src/main/java/com/ibm/fhir/model/ucum/util/UCUMUtil.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * (C) Copyright IBM Corp. 2020
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.model.ucum.util;
+
+import org.antlr.v4.runtime.ANTLRErrorListener;
+import org.antlr.v4.runtime.BaseErrorListener;
+import org.antlr.v4.runtime.CharStreams;
+import org.antlr.v4.runtime.CommonTokenStream;
+import org.antlr.v4.runtime.RecognitionException;
+import org.antlr.v4.runtime.Recognizer;
+import org.antlr.v4.runtime.misc.ParseCancellationException;
+
+import com.ibm.fhir.model.ucum.UCUMLexer;
+import com.ibm.fhir.model.ucum.UCUMParser;
+import com.ibm.fhir.model.ucum.UCUMParser.MainTermContext;
+
+/**
+ * Utility class for UCUM.
+ */
+public class UCUMUtil {
+
+    private static final ANTLRErrorListener SYNTAX_ERROR_LISTENER = new BaseErrorListener() {
+
+        @Override
+        public void syntaxError(Recognizer<?, ?> recognizer, Object offendingSymbol, int line, int charPositionInLine, String msg, RecognitionException e) {
+            throw new ParseCancellationException(String.format("line %d:%d %s", line, charPositionInLine, msg), e);
+        }
+    };
+
+    private UCUMUtil() {
+    }
+
+    /**
+     * Determines if the string is a valid UCUM string.
+     * 
+     * @param ucumString
+     *            the string
+     * @return true or false
+     */
+    public static boolean isValidUcum(String ucumString) {
```

Comment:
Not familiar with antlr lexer/parser, so don't know if it's a big deal or not - is there not a way to avoid instantiating a new lexer/parser every time you want to parse a string?

Reply:
I followed the same flow as FHIRPathUtil.java, which instantiated a new lexer/parser each time, and that appears to be how it needs to work.